dspace-server-webapp/src/test/java/org/dspace/app/rest/RequestItemRepositoryIT.java

```diff
@@ -0,0 +1,393 @@
+     * Test of createAndReturn method, with an UNauthenticated user.
+     * This should succeed:  anyone can file a request.
+     *
+     * @throws java.sql.SQLException passed through.
+     * @throws org.dspace.authorize.AuthorizeException passed through.
+     * @throws java.io.IOException passed through.
+     */
+    @Test
+    public void testCreateAndReturnNotAuthenticated()
+            throws SQLException, AuthorizeException, IOException, Exception {
+        System.out.println("createAndReturn");
+
+        // Create some necessary objects.
+        context.turnOffAuthorisationSystem();
+
+        Collection col = CollectionBuilder.createCollection(context,
+                parentCommunity).build();
+        Item item = ItemBuilder.createItem(context, col).build();
+        InputStream is = new ByteArrayInputStream(new byte[0]);
+        Bitstream bitstream = BitstreamBuilder.createBitstream(context, item, is)
+                .withName("/dev/null")
+                .withMimeType("text/plain")
+                .build();
+
+        context.restoreAuthSystemState();
+
+        // Fake up a request in REST form.
+        RequestItemRest rir = new RequestItemRest();
+        rir.setBitstreamId(bitstream.getID().toString());
+        rir.setItemId(item.getID().toString());
+        rir.setRequestEmail(RequestItemBuilder.REQ_EMAIL);
+        rir.setRequestMessage(RequestItemBuilder.REQ_MESSAGE);
+        rir.setRequestName(RequestItemBuilder.REQ_NAME);
+        rir.setAllfiles(false);
+
+        // Create it and see if it was created correctly.
+        ObjectMapper mapper = new ObjectMapper();
+        getClient().perform(post(URI_ROOT)
+                        .content(mapper.writeValueAsBytes(rir))
+                        .contentType(contentType))
+                .andExpect(status().isCreated())
+                .andExpect(content().contentType(contentType))
+                .andExpect(jsonPath("$", Matchers.allOf(
+                        hasJsonPath("$.id", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.type", is(RequestItemRest.NAME)),
+                        hasJsonPath("$.token", not(is(emptyOrNullString()))),
+                        hasJsonPath("$.requestEmail", is(RequestItemBuilder.REQ_EMAIL)),
+                        hasJsonPath("$.requestMessage", is(RequestItemBuilder.REQ_MESSAGE)),
+                        hasJsonPath("$.requestName", is(RequestItemBuilder.REQ_NAME)),
+                        hasJsonPath("$.allfiles", is(false)),
+                        hasJsonPath("$.requestDate", not(is(emptyOrNullString()))), // TODO should be an ISO datetime
+                        hasJsonPath("$._links.self.href", not(is(emptyOrNullString())))
+                )))
+                .andDo((var result) -> saveToken(read(result.getResponse().getContentAsString(), "token")))
+                .andReturn();
+
+        // Clean up the created request.
+        RequestItemBuilder.deleteRequestItem(requestToken);
```

Comment:
@mwoodiupui : I _think_ the issue is with your `andDo()` syntax.  I'm not sure this this syntax is correct -- and it may result in some ItemRequests not being deleted, which seems to be the root issue. 

Other examples in our codebase use this syntax instead...
```
AtomicReference<String> idRef = new AtomicReference<>();
getClient().perform(post(
....
.andDo(result -> idRef.set(read(result.getResponse().getContentAsString(), "token")));
```
Then, you get the value out of the `idRef` variable like this...
```
// I'd recommend doing a try/finally to ensure this deletion ALWAYS runs
} finally {
    RequestItemBuilder.deleteRequestItem(idRef.get());
}
```

Essentially, my suspicion here is that this cleanup isn't working as you expected. I'd bet that `requestToken` ends up not being the value you expect and something is not being deleted.

Reply:
More background info on my suspicions... in checking the test failures, I saw the **first** failure came out of your `testCreateAndReturnAuthenticated` test method.  It appeared to show that the CollectionBuilder was trying to cleanup an Item, and that Item couldn't be deleted cause it was still related to an ItemRequest.

So, I started looking at ItemRequest deletion, as that sort of failure implies that one (or more) of those requests are "hanging around" even after the RequestItemBuilder ran it's cleanup task.

When these sorts of failures occur, I find that the **first** failure is _usually_ the most significant one...the remainder are usually just side effects of the first one.

Reply:
I made your suggested changes.  Tests still broke.  Other than atomicity, I think my code was doing the same thing as the suggestion, but at least now mine looks familiar if you know the others.

The actual problem was that RequestItemService.deleteRequestItem has its own Context and did not commit its work.

Reply:
Good catch then on finding the real problem!  Sorry if I misunderstood what your code was doing...it just wasn't the normal syntax I was familiar with.